### ISSUE TITLE
[🐛 Bug]: browser.options is missing most properties

### ISSUE THREAD

**Issue by therealbrad** [webdriverio/webdriverio#11573]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.20.4

### Node.js Version

18.16.0

### Mode

WDIO Testrunner

### What happened?

I am attempting to get the array of defined services via `browser.options.services` which was working in 8.16.22. I updated to 8.20.4 and browser.options is missing lost of properties including reporters, services, suites. Here is my browser.options in 8.20.4:

```{
    "protocol": "http",
    "hostname": "0.0.0.0",
    "path": "/",
    "capabilities":
    {
        "browserName": "chrome",
        "goog:chromeOptions":
        {
            "binary": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
            "args":
            [
                "start-maximized",
                "disable-infobars",
                "ignore-gpu-blacklist",
                "test-type",
                "disable-gpu",
                "--remote-allow-origins=*",
                "--use-fake-ui-for-media-stream",
                "--use-fake-device-for-media-stream"
            ]
        }
    },
    "logLevel": "warn",
    "connectionRetryTimeout": 600000,
    "connectionRetryCount": 4,
    "logLevels":
    {},
    "enableDirectConnect": true,
    "strictSSL": true,
    "cacheDir": "/var/folders/pq/2kcp85fx2_b4xrbdtlvkn6bc0000gn/T",
    "port": 58671,
    "requestedCapabilities":
    {
        "browserName": "chrome",
        "goog:chromeOptions":
        {
            "binary": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
            "args":
            [
                "start-maximized",
                "disable-infobars",
                "ignore-gpu-blacklist",
                "test-type",
                "disable-gpu",
                "--remote-allow-origins=*",
                "--use-fake-ui-for-media-stream",
                "--use-fake-device-for-media-stream"
            ]
        }
    },
    "automationProtocol": "webdriver",
    "baseUrl": "<REDACTED>",
    "waitforInterval": 500,
    "waitforTimeout": 10000,
    "onReload":
    [],
    "beforeCommand":
    [],
    "afterCommand":
    []
}```

### What is your expected behavior?

browser.options should include all the properties like this from 8.16.22:
```{
    "protocol": "http",
    "hostname": "0.0.0.0",
    "path": "/",
    "capabilities":
    {
        "browserName": "chrome",
        "goog:chromeOptions":
        {
            "binary": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
            "args":
            [
                "start-maximized",
                "disable-infobars",
                "ignore-gpu-blacklist",
                "test-type",
                "disable-gpu",
                "--remote-allow-origins=*",
                "--use-fake-ui-for-media-stream",
                "--use-fake-device-for-media-stream"
            ]
        }
    },
    "logLevel": "warn",
    "connectionRetryTimeout": 600000,
    "connectionRetryCount": 4,
    "logLevels":
    {},
    "enableDirectConnect": true,
    "strictSSL": true,
    "cacheDir": "/var/folders/pq/2kcp85fx2_b4xrbdtlvkn6bc0000gn/T",
    "port": 58671,
    "requestedCapabilities":
    {
        "browserName": "chrome",
        "goog:chromeOptions":
        {
            "binary": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
            "args":
            [
                "start-maximized",
                "disable-infobars",
                "ignore-gpu-blacklist",
                "test-type",
                "disable-gpu",
                "--remote-allow-origins=*",
                "--use-fake-ui-for-media-stream",
                "--use-fake-device-for-media-stream"
            ]
        }
    },
    "automationProtocol": "webdriver",
    "specs":
    [
        "../test/specs/**/*.js"
    ],
    "exclude":
    [],
    "suites":
    {
        "admin":
        [
            "../test/specs/admin/**/*.js"
        ],
        "analytics":
        [
            "../test/specs/analytics/**/*.js"
        ],
        "chromeextension":
        [
            "../test/specs/chrome_extension/**/*.js"
        ],
        "ci":
        [
            "../test/specs/ci/**/*.js"
        ],
        "web":
        [
            "../test/specs/web/**/*.js"
        ],
        "prod":
        [
            "../test/specs/prod/**/*.js"
        ],
        "refract":
        [
            "../test/specs/refract/**/*.js"
        ]
    },
    "rootDir": "/Users/bderman/git/wdio/web/config",
    "baseUrl": "<REDACTED>",
    "bail": 10,
    "waitforInterval": 500,
    "waitforTimeout": 10000,
    "framework": "mocha",
    "reporters":
    [
        "spec",
        [
            "junit",
            {
                "outputDir": "./reports"
            }
        ],
        [
            "allure",
            {
                "outputDir": "reports/allure-results",
                "disableWebdriverStepsReporting": true,
                "disableWebdriverScreenshotsReporting": false,
                "reportedEnvironmentVars": true,
                "tmsLinkTemplate": "<REDACTED>"
            }
        ]
    ],
    "services":
    [
        [
            "cleanuptotal"
        ],
        [
            "performancetotal",
            {
                "dropResultsFromFailedTest": true,
                "performanceResultsDirectory": "web/reports/performance",
                "performanceResultsFileName": "performance-results"
            }
        ],
        [
            "image-comparison",
            {
                "baselineFolder": "/Users/bderman/git/wdio/web/test/imageBaseline/",
                "formatImageName": "{tag}-{logName}-{width}x{height}",
                "screenshotPath": "/Users/bderman/git/wdio/web/.tmp/",
                "savePerInstance": true,
                "autoSaveBaseline": true,
                "blockOutStatusBar": true,
                "blockOutToolBar": true,
                "isHybridApp": false,
                "tabbableOptions":
                {
                    "circle":
                    {
                        "size": 18,
                        "fontSize": 18
                    },
                    "line":
                    {
                        "color": "#ff221a",
                        "width": 3
                    }
                }
            }
        ]
    ],
    "execArgv":
    [],
    "maxInstances": 1,
    "maxInstancesPerCapability": 100,
    "injectGlobals": true,
    "specFileRetries": 0,
    "specFileRetriesDelay": 0,
    "specFileRetriesDeferred": false,
    "filesToWatch":
    [],
    "onPrepare":
    [
        null
    ],
    "onWorkerStart":
    [],
    "onWorkerEnd":
    [],
    "before":
    [
        null,
        null,
        null,
        null
    ],
    "beforeSession":
    [
        null
    ],
    "beforeSuite":
    [
        null
    ],
    "beforeHook":
    [],
    "beforeTest":
    [
        null
    ],
    "beforeCommand":
    [],
    "afterCommand":
    [],
    "afterTest":
    [
        null,
        null,
        null
    ],
    "afterHook":
    [],
    "afterSuite":
    [],
    "afterSession":
    [],
    "after":
    [
        null,
        null
    ],
    "onComplete":
    [
        null
    ],
    "onReload":
    []
}```

### How to reproduce the bug.

Simply log browser.options to see missing properties.

### Relevant log output

```typescript
Output is normal between the versions aside from the inability to get the missing properties from browser.options.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

We have moved the way we attach and set instance options in the recent version which caused this to happen. I suggest to two the following:

- add a generic type to the `options` property that changes based on whether someone is using standalone or testrunner so that folks using TS in standalone they get a limited typed options object
- within the testrunner (I assume `@wdio/runner`) we should enhance the options object of the browser instance with the testrunner options which include things like `spec`, `services` and `reporter`.

We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by christian-bromann** [comment]
Fixed in #11589
